### PR TITLE
changing vtsearch.py to use requests, adding 'allinfo' option

### PR DESCRIPTION
**Pull request by mjglanzer**

### CODE REVIEW

**Comment by Drewsif** [comment]
Is there a limit to the amount of files you can submit in one query still? The old version had a limit of 25 per request

File: modules/Antivirus/vtsearch.py [hunk 1]
```diff
@@ -32,66 +26,84 @@ def check(conf=DEFAULTCONF):
...
-        rotkey = _repeatlist(conf['apikey'])
+    if isinstance(apikey, list):
+        rotkey = _repeatlist(apikey)
         apikey = rotkey.next()
-    results = []
-    md5s, junk = REQUIRES[0]
-    requests = [[]]
-    md5name = {}
-    for fname, md5 in md5s:
-        if len(requests[-1]) == 25:
```
Comment (mjglanzer) [inline modules/Antivirus/vtsearch.py]: Strange that I missed that hash-per-query limit check from the old code given its thorough commenting :) 

I've updated the PR to break all the hashes into <= 25 hash sub-lists.

Comment (Drewsif) [reply modules/Antivirus/vtsearch.py]: Comments are hard! I'll test it out

**Comment by awest1339** [comment]
* Doesn't run on my install
* https://github.com/mjglanzer/multiscanner/blob/6c57a77cc052cadf240568ecfab50e08a80b3aa7/modules/Antivirus/vtsearch.py#L16
  * DEFAULTCONF needs to have the "allinfo" key in there, otherwise the default config generated will throw an error on line 29
    * (I'd also recommend using a .get() for line 29 instead of key access)

**Comment by mjglanzer** [comment]
- Line 55 is supposed to be '+' - I want to concatenate the lists. What is the error you are getting?

- I'll add 'allinfo' to DEFAULTCONF.

**Comment by awest1339** [comment]
.json() returns a dict, not a list

**Comment by mjglanzer** [comment]
requests .json() can return a list of json/dict content

**Comment by awest1339** [comment]
Have you run the code? :) In this use, it returns a dict in both python 2 and 3.

**Comment by mjglanzer** [comment]
Yeah, but still... :)  That is with multiple hashes?  I don't have a VT key on hand - I'll look at it tomorrow.

**Comment by awest1339** [comment]
With more than one file, the module performs as expected. with one file, the module throws an error about concatenating a list and a dict since with one hash, it gets a dict back

**Comment by mjglanzer** [comment]
Dammit, Austin - That's what I had said initially - 

"I had been testing by sending multiple hashes - I bet the error happens when you send a single hash and a dict (not a list) is returned. I'll fix."

**Comment by awest1339** [comment]
lol i misread your error 🗡

**Comment by mjglanzer** [comment]
:)

**Pull request merged**